packages/esm-patient-registration-app/src/patient-registration/input/custom-input/identifier/utils.test.ts

```diff
@@ -0,0 +1,81 @@
+import { isUniqueIdentifierTypeForOffline, shouldBlockPatientIdentifierInOfflineMode } from './utils';
+
+interface IdentifierTypeOptions {
+  uniquenessBehavior?: 'UNIQUE' | 'LOCATION' | 'NON_UNIQUE';
+  manualEntryEnabled?: boolean;
+  automaticGenerationEnabled?: boolean;
+}
+
+function createIdentifierType(options: IdentifierTypeOptions) {
+  return {
+    uniquenessBehavior: (options.uniquenessBehavior as 'UNIQUE' | 'LOCATION' | 'NON_UNIQUE') || null,
+    identifierSources: [
+      {
+        uuid: 'identifier-source-uuid',
+        name: 'Identifier Source Name',
+        autoGenerationOption: {
+          manualEntryEnabled: (options.manualEntryEnabled as boolean) || false,
+          automaticGenerationEnabled: (options.automaticGenerationEnabled as boolean) || true,
```

Comment:
```suggestion
          manualEntryEnabled: (options.manualEntryEnabled) || false,
          automaticGenerationEnabled: (options.automaticGenerationEnabled) || true,
```
Do we still need to infer the types @ayush-AI even the interface is defined?